cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -102,31 +106,37 @@ abstract class DataprocClient implements AutoCloseable {
   private final ClusterControllerClient client;
   private final ComputeFactory computeFactory;
   private Compute compute;
+  private ErrorCategory errorCategory;
 
   protected DataprocClient(DataprocConf conf, ClusterControllerClient client,
-      ComputeFactory computeFactory) {
+      ComputeFactory computeFactory, ErrorCategory errorCategory) {
     this.conf = conf;
     this.client = client;
     this.computeFactory = computeFactory;
+    this.errorCategory = errorCategory;
   }
 
   private static String findNetwork(Compute compute, String project)
-      throws IOException, RetryableProvisionException {
+      throws RetryableProvisionException {
     List<Network> networks;
     try {
       NetworkList networkList = compute.networks().list(project).execute();
       networks = networkList.getItems();
     } catch (Exception e) {
       handleRetryableExceptions(e);
-      throw new DataprocRuntimeException(e);
+      String errorReason = String.format("Unable list networks in project '%s'", project);
+      if (e instanceof GoogleJsonResponseException) {
+        throw handleGoogleJsonResponseException((GoogleJsonResponseException) e, errorReason);
+      }
+      throw getDataprocRuntimeException(DataprocRuntimeException.ERROR_CATEGORY_CONFIGURATION,
```

Comment:
Please create error category based on proper PROVISIONING/DEPROVISIONING data in `errorCategory` field. Ditto in other places below

Reply:
done.